src/rebar_prv_dialyzer.erl

```diff
@@ -193,9 +194,20 @@ do_update_proj_plt(State, Plt, Output) ->
     end.
 
 proj_plt_files(State) ->
-    BasePltApps = base_plt_apps(State),
+    {BasePltApps, BasePltMods} =
+        case is_update_base_plt_active(State) of
+            false ->
+                {ok, BaseFiles} = plt_files(get_base_plt(State)),
+                BaseMods =
+                    lists:map(fun(File) ->
+                                      list_to_atom(filename:basename(File, code:objfile_extension()))
+                              end, BaseFiles),
+                {[], BaseMods};
+            true  ->
+                {base_plt_apps(State), get_config(State, base_plt_mods, [])}
+        end,
+
     PltApps = get_config(State, plt_extra_apps, []) ++ BasePltApps,
-    BasePltMods = get_config(State, base_plt_mods, []),
```

Comment:
We always want to build the same project PLT, no matter what the contents of the base PLT. Therefore we don't want to infer anything from the contents of the base PLT. When building the project PLT, we copy the base PLT and then do a diff with the desired project PLT to find out what changes need to be madet. 

These are extra modules we want added to the base PLT, and also to the project PLT. Even if these modules are not actually in the base PLT, we want the configured list to be in the project PLT so we always build the same project PLT, no matter what the state of any of the PLTs.